Tweaks (#18)
viam-labs/hand_raiser: README.md

# Hand Raiser robot for Zoom meetings

When we have large meetings, sometimes the remote employees are unable to participate in the Q&A section at the end, even though they're able to view the main meeting over Zoom. This repo provides a way for a robot to join the Zoom meeting and raise a physical hand in the (meatspace) meeting room when someone on Zoom has a question.

## Installation

1. Clone this repo locally.
2. Run `pip install -r requirements.txt` to install the dependencies.
3. Edit `secrets.py` so it contains the robot's secret and URL. You can get these from anyone who worked on this project.
4. Step 3 probably made the repo dirty. Run `git update-index --skip-worktree secrets.py` to make the repo clean again.

## Running
1. When you want Hand Raiser Bot to join a meeting, run `./main.py '<url-of-zoom-meeting>'` to start it. The URL likely contains a question mark, so we recommend enclosing the entire URL in single quotes so your terminal doesn't try pattern-matching on it.

> 2. This will open a Chrome window and join the Zoom meeting as the user "Hand Raiser Bot".

Reviewer: ```suggestion
2. This will open a Chrome window and join the Zoom meeting as the user "Hand Raiser Bot."
```
[opt/nit]

> 3. Whenever someone in the Zoom meeting selects the "Raise Hand" reaction, we'll move the servo on the robot. It will be raised whenever _at least 1_ person in the Zoom meeting has their hand raised, and lowered again when no one has their hand raised.

Reviewer: ```suggestion
3. Whenever someone in the Zoom meeting selects the "Raise Hand" reaction, the servo on the robot moves. The hand will be raised whenever _at least 1_ person in the Zoom meeting has their hand raised, and lowered again when no one has their hand raised.
```

> 4. If Hand Raiser Bot's servo has been raised for at least 30 seconds, we will wiggle it side to side, to try to call attention to ourselves.

Reviewer: ```suggestion
4. If the hand has been raised for at least 30 seconds, it will begin to wiggle side-to-side at intervals to try to gain attention.
5. Once a Zoom participant has been called upon, they should lower their hand in the Zoom interface! This will lower the robot hand, unless other meeting participants also have raised hands in Zoom.
```

> 5. When the meeting is over (or when you want Hand Raiser Bot to leave), hit control-C in the terminal to shut everything down. This will also lower the servo even if someone in the Zoom meeting still has their hand raised.

Reviewer: ```suggestion
6. When the meeting is over (or when you want Hand Raiser Bot to leave), hit control-C in the terminal to shut everything down. This will also lower the servo even if someone in the Zoom meeting still has their hand raised.
```
(if above suggestion is implemented)

## Code Layout

> - `robot.py` is how to talk to Viam to move the hardware itself. This can raise and lower the servo, and wiggle it if it has been raised for long enough.

Reviewer: ```suggestion
- `robot.py` talks to `viam-server` to move the hardware itself. This script raises and lowers the servo, and wiggles it if it has been raised for a while.
```
[opt]

Author: I like this except the word "script" is misplaced: this file acts as a library to be used elsewhere, and main.py is the "real" script. 

I'm new to suggested changes within a code review: how do I tweak your suggested change without just making the change on my laptop and then committing/pushing it up here?

Reviewer: It is possible to edit peoples' suggestions but I try to avoid doing that since it gets confusing. Instead, you can just make your own new suggestion in this thread, copy-paste the contents of my suggestion into it, and comment that then commit it. IDK if this is right either:
```suggestion
- `robot.py` talks to `viam-server` to move the hardware itself. The code in this file raises and lowers the servo, and wiggles it if it has been raised for a while.
```

> - `audience.py` is how to keep track of how many hands are raised. This will tell the robot when it's time to raise and lower the hand.

Reviewer: ```suggestion
- `audience.py` keeps track of how many hands are raised. This tells the robot when it's time to raise and lower the hand.
```
[opt]

> - `zoom_monitor.py` is how to use Selenium to open a web browser and join the Zoom meeting. It lets you count how many participants in the meeting have their hands raised.

Reviewer: ```suggestion
- `zoom_monitor.py` uses Selenium to open a web browser and join the Zoom meeting. It counts how many participants in the meeting have their hands raised.
```
[opt]
